### DISCUSSION TITLE
AssertionError: cycle-free graph violated: partition depends on itself

### DISCUSSION THREAD

**Discussion by btwbtw01** [Xilinx/finn#567] [Q&A]
Hi, I'm following the end2end_cnv toolflow with vgg16 model that I modified. However, during partitioning I run into the following error while creating a dataflow partition
![image](https://user-images.githubusercontent.com/85877169/156871141-482a3fef-7cb0-498a-ab16-57cb0ac8f46b.png)
Here is my model before streamling and after streamling
![end2end_vgg16_pre_post](https://user-images.githubusercontent.com/85877169/156871224-b789438f-30c0-4804-9a64-ccb4152d4412.png)
![end2end_vgg16_streamlined](https://user-images.githubusercontent.com/85877169/156871226-5e847dd5-2893-4fed-817e-4232f8112ed9.png)

**Comment by nhma20** [comment]
https://github.com/Xilinx/finn/discussions/431 or https://github.com/Xilinx/finn/discussions/495 may be what you are looking for

**Comment by btwbtw01** [comment]
Hi, @nhma20 
Thanks for your reply. It works, but I face another problems. When I convert the mobilenet model to hls, the MatMul node didn't remove and cause error. Do you know how to solve it? Does anybody know how to solve it?
![end2end_mobilenet_hls_layers](https://user-images.githubusercontent.com/85877169/157187829-1ad27842-113d-4a19-ae44-b9300cd0bde0.png)
